projects/plugins/crm/includes/ZeroBSCRM.DAL3.php

```diff
@@ -571,6 +571,57 @@ public function fields_to_hide_on_frontend( $obj_type_int = false ){
 
     }
 
+	/**
+	 * Returns a count of objects of a type which are associated with a company or contact
+	 * Supports Quotes, Invoices, Transactions, Events currently
+	 *
+	 * @param int $cust_id The company / contact ID.
+	 * @param int $obj_type_id The type constant being checked (eg ZBS_TYPE_QUOTE).
+	 * @param int $zbs_type The contact or company type, for example ZBS_TYPE_COMPANY, ZBS_TYPE_CONTACT (default contact).
+	 *
+	 * @return int The count of relevant objects of the given type.
+	 */
+	public function customer_has_count_obj_type( $cust_id, $obj_type_id, $zbs_type = ZBS_TYPE_CONTACT ) {
+		// phpcs:disable WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+		global $ZBSCRM_t;
+		global $wpdb;
+
+		switch ( $obj_type_id ) {
+			case ZBS_TYPE_QUOTE:
+				$table_name = $ZBSCRM_t['quotes'];
+				break;
+
+			case ZBS_TYPE_INVOICE:
+				$table_name = $ZBSCRM_t['invoices'];
+				break;
+
+			case ZBS_TYPE_TRANSACTION:
+				$table_name = $ZBSCRM_t['transactions'];
+				break;
+
+			case ZBS_TYPE_EVENT:
+				$table_name = $ZBSCRM_t['events'];
+				break;
+
+			default:
+				// For any unsupported objtype.
+				return -1;
+		}
+
+		$obj_query = 'SELECT COUNT(obj_table.id) FROM ' . $table_name . ' obj_table'
```

Comment:
This is legacy code, but at first I was wondering why there's a join on a given table. In the end it's probably so orphan references are ignored.